js/player_thread.js

```diff
@@ -0,0 +1,475 @@
+/* eslint-disable no-underscore-dangle */
+/* global navigator MediaMetadata */
+/* global Howl Howler */
+
+{
+  const sendEvent = (event) => {
+    (chrome || browser).runtime.sendMessage(event);
+  };
+
+  /**
+   * Player class containing the state of our playlist and where we are in it.
+   * Includes all methods for playing, skipping, updating the display, etc.
+   * @param {Array} playlist Array of objects with playlist song details ({title, file, howl}).
+   */
+  class Player {
+    constructor() {
+      this.playlist = [];
+      this.index = -1;
+      this._loop_mode = 0;
+      this._media_uri_list = {};
+      this.playedFrom = 0;
+    }
+
+    setRefreshRate(rate = 10) {
+      clearInterval(this.refreshTimer);
+      this.refreshTimer = setInterval(() => {
+        if (this.playing) {
+          this.sendFrameUpdate();
+        }
+      }, 1000 / rate);
+    }
+
+    get currentAudio() {
+      return this.playlist[this.index];
+    }
+
+    get currentHowl() {
+      return this.currentAudio && this.currentAudio.howl;
+    }
+
+    get playing() {
+      return this.currentHowl ? this.currentHowl.playing() : false;
+    }
+
+    static get muted() {
+      return !!Howler._muted;
+    }
+
+    insertAudio(audio, idx) {
+      const audioData = {
+        ...audio,
+        disabled: false, // avoid first time load block
+        howl: null,
+      };
+      if (idx) {
+        this.playlist.splice(idx, 0, [audio]);
+      } else {
+        this.playlist.push(audioData);
+      }
+      this.sendPlaylistEvent();
+    }
+
+    removeAudio(idx) {
+      if (!this.playlist[idx]) {
+        return;
+      }
+      if (this.playlist[idx].howl && this.playlist[idx].howl.playing()) {
+        this.skip('next');
+      }
+      this.playlist.splice(idx, 1);
+      this.sendPlaylistEvent();
+    }
+
+    appendAudioList(list) {
+      if (!Array.isArray(list)) {
+        return;
+      }
+      list.forEach((audio) => {
+        this.insertAudio(audio);
+      });
+    }
+
+    clearPlaylist() {
+      this.playlist = [];
+      Howler.stop();
+      this.sendPlaylistEvent();
+    }
+
+    setNewPlaylist(list) {
+      if (list.length) {
+        Howler.stop();
+
+        this.playlist = list.map(audio => ({
+          ...audio,
+          howl: null,
+        }));
+        this.load(0);
+      }
+      this.sendPlaylistEvent();
+    }
+
+    playById(id) {
+      const idx = this.playlist.findIndex(audio => audio.id === id);
```

Comment:
//check if same song is already playing, if so, seek to start of song and play, and it will fix same song play current for several times bug
      if(idx === this.index){
        this.seek(0);
        this.currentHowl.play();
        return;
      }

Reply:
Modification is applied in `finishLoad` function.